src/main/java/by/exadel/internship/controller/InternshipController.java

```diff
@@ -31,8 +33,19 @@
 
     @GetMapping("/{internshipId}")
     @ApiOperation("return internship by id")
-    public GuestFullInternshipDTO getInternshipById(@PathVariable("internshipId") UUID internshipId) {
-        return internshipService.getById(internshipId);
+    public ResponseEntity<?> getInternshipById(@PathVariable("internshipId") UUID internshipId, boolean isAdminInformation) {
+
+        ResponseEntity responseEntity;
+
+        if (isAdminInformation) {
```

Comment:
Object dto = isAdminInformation ? internshipService.getUserInternshipDTOById(internshipId) : internshipService.getGuestFullInternshipDTOById(internshipId);
return new ResponseEntity<>(dto, HttpStatus.OK);

Reply:
Take a look may be one DTO could be extended from another. I believe full version will have all fields from the limited one. In this case Use limited type instead of Object

Reply:
aaaa!  very cool, thank you! done

Reply:
> Take a look may be one DTO could be extended from another. I believe full version will have all fields from the limited one. In this case use limited type instead of Object

I tried to do this, but I got a problem with the mapper, we already have two DTO's with extending, but this would be the third level, with which the mapper fundamentally disagreed

Reply:
done, i extrends UserInternshipDto from GuestShortInternshipDTO, it solved this problem.